examples/dreambooth/train_dreambooth.py

```diff
@@ -616,16 +634,41 @@ def main(args):
     logger.info(f"  Total train batch size (w. parallel, distributed & accumulation) = {total_batch_size}")
     logger.info(f"  Gradient Accumulation steps = {args.gradient_accumulation_steps}")
     logger.info(f"  Total optimization steps = {args.max_train_steps}")
+    global_step = 0
+    starting_epoch = 0
+
+    if args.resume_from_checkpoint:
+        if args.resume_from_checkpoint != "latest":
+            path = os.path.basename(args.resume_from_checkpoint)
+        else:
+            # Get the mos recent checkpoint
+            dirs = os.listdir(args.output_dir)
+            dirs = [d for d in dirs if d.startswith("checkpoint")]
+            dirs = sorted(dirs, key=lambda x: int(x.split("-")[1]))
+            path = dirs[-1]
+        accelerator.print(f"Resuming from checkpoint {path}")
+        accelerator.load_state(os.path.join(args.output_dir, path))
+        global_step = int(path.split("-")[1])
+
+        resume_step = global_step * args.gradient_accumulation_steps
+        starting_epoch = resume_step // num_update_steps_per_epoch
+        resume_step = resume_step % num_update_steps_per_epoch
```

Comment:
These adjustments can be confusing, in my opinion it doesn't make much sense to use the concept of "epoch" for things like Dreambooth. But it may be important to keep the same structure for consistency with other training scripts.

Reply:
Fine for me. We could maybe do some renaming here: 

```suggestion
        resume_global_step = global_step * args.gradient_accumulation_steps
        resume_epoch = resume_global_step // num_update_steps_per_epoch
        resume_step = resume_global_step % num_update_steps_per_epoch
```

but overall this is fine for me! Agree that the concept of "epoch" is weird for most dreambooth cases, but it was there before and there are cases where it could make sense if a lot of data is available IMO.

Reply:
I think we can rename `resume_global_step` as suggested, but I'd rather keep `starting_epoch` or `first_epoch` because I think it's clearer when it's used in the loop here (the default value when we are not resuming is `0`, of course): https://github.com/huggingface/diffusers/pull/1668/files#diff-8702f762e46a3b5363085930b0b045de554909d32560864031ca7b12ddd349d5R661